src/base/asyncfilestorage.h

```diff
@@ -24,35 +24,38 @@
  * modify file(s), you may extend this exception to your version of the file(s),
  * but you are not obligated to do so. If you do not wish to do so, delete this
  * exception statement from your version.
- *
- * Contact : chris@qbittorrent.org
  */
 
-#ifndef RSSSETTINGSDLG_H
-#define RSSSETTINGSDLG_H
-
-#include <QDialog>
+#pragma once
 
-QT_BEGIN_NAMESPACE
-namespace Ui {
-    class RssSettingsDlg;
-}
-QT_END_NAMESPACE
+#include <QDir>
+#include <QFile>
+#include <QHash>
+#include <QMutex>
+#include <QObject>
 
-class RssSettingsDlg : public QDialog
+class AsyncFileStorage: public QObject
 {
     Q_OBJECT
+    explicit AsyncFileStorage(const QDir &storageDir, QFile *lockFile, QObject *parent = nullptr);
 
 public:
-    explicit RssSettingsDlg(QWidget *parent = 0);
-    ~RssSettingsDlg();
+    static AsyncFileStorage *create(const QString &storageFolder, QObject *parent = nullptr);
+    ~AsyncFileStorage() override;
+
+    void store(const QString &fileName, const QByteArray &data);
 
-protected slots:
-    void on_buttonBox_accepted();
+    QDir storageDir() const;
+
+signals:
+    void failed(const QString &fileName, const QString &errorString);
 
 private:
-    Ui::RssSettingsDlg *ui;
+    void processPendingJobs();
 
+    QDir m_storageDir;
+    QFile *m_lockFile;
+    QHash<QString, QByteArray> m_jobs;
+    QMutex m_jobsMutex;
+    bool m_isIdle = true;
```

Comment:
>Just so you know, this design decision will result in recompilations of stuff that include this header in case we want to change the value of the variable. 

But this is only the default value! It does not change practically never. In any case, much less than the other members of the class.